connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -43,10 +43,38 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeoutException;
 
-public class RestClient {
+public class RestClient implements AutoCloseable {
     private static final Logger log = LoggerFactory.getLogger(RestClient.class);
     private static final ObjectMapper JSON_SERDE = new ObjectMapper();
 
+    private final HttpClient client;
+    public RestClient(WorkerConfig config) {
+        client = new HttpClient(SSLUtils.createClientSideSslContextFactory(config));
```

Comment:
I see the note in the description that this should be a lateral change, but that appears to only cover the case of an SSL-configured client being used to make plain HTTP requests.

What about cases where the user has not specified any SSL-related properties in their worker config? Will `SSLUtils::createClientSideSslContextFactory` essentially create a no-op factory that can still be used for HTTP requests?

If it's not too painful, it'd be nice to see some test coverage for the `RestClient` with/without SSL properties in the `WorkerConfig` it's instantiated with.

Reply:
This new integration test is fantastic, thanks!

Reply:
It's even better now: It actually works!

Also it tests all phases of the roll from a non-SSL to SSL cluster.

Reply:
Hmm we are running into a wierd case because of this where we have ssl configs configured however we set the security.protocol to `SASL_PLAINTEXT` and ssl still makes an apearence (and fails our test because our test is a bit of a ridiculous configuration where we set the keystore location value for ssl however we dont create it as we arnt running with ssl).  IDK if one should consider the `security.protocol` here but anyway....

Reply:
@imcdo If this change has (or appears to have) broken anything, feel free to file a Jira ticket and add example worker configs, testing scenarios, and anything else that can help us understand what's going wrong and, if applicable, how to fix it.

Reply:
I dont feel like I have enough background to complain if its really a breaking change or not - its just a change in expectations so i wouldnt worry about it.  I can still open the jira if you are interested in the scenario itself if you think it might be a valid issue.

Reply:
https://issues.apache.org/jira/browse/KAFKA-14816